src/snowflake/snowpark/modin/pandas/groupby.py

```diff
@@ -1190,6 +1212,9 @@ def is_monotonic_increasing(self):
             name="is_monotonic_increasing", class_="GroupBy"
         )
 
+    def size(self):
+        return super().size().rename(self._df.columns[-1])
```

Comment:
I thought we already handled the name in this part 
```
elif isinstance(self._df, Series):
            result.name = self._df.name
```

Reply:
Something wacky is happening in the frontend and the `isinstance` check always evaluates to `False`. It is likely that Snowpark pandas is doing a `reset_index` before the `groupby` so this is needed currently. It can be removed once we fix `Series.groupby` to match Modin.

Reply:
this seems checking the frontend self._df, Do you mean we did a reset_index at frontend? that is weird, do you know where we did this?

Reply:
I couldn't find where in a cursory look, but I didn't want to block the customer on debugging this. It will get resolved once we align with Modin OSS.

Reply:
Our current release is already out, the next release will come in next 1 or 2 weeks, so i don't see a reason to hurry for this. I don't want to relay on align with Modin OSS for this, it seems still far away, if there is something wrong we did, we can fix it right away, so it doesn't cause us other problem when we implement other methods

Reply:
This is not the only request from the customer, and I would like for our team to be in the habit of shipping quickly. This works, debugging this will not add any value to the user, and fixing this is planned for the future. I prefer not to block the PR on something like this, since it may force a big change that affects every other groupby method. IMO it's better to do that when we are migrating code.

Reply:
We can ship quickly. However, ship quickly doesn't really mean ship hacky code without knowing the root cause, this will make our debugging difficult if we run into issue, and made our code base worse and worse. As far as we know the root cause, we can have follow up tickets to do it right after, they migration doesn't seem going to happen anytime soon also.

Reply:
In the interest of unblocking this, I have root caused the issue and filed a ticket: https://snowflakecomputing.atlassian.net/browse/SNOW-1478924

The migration is in progress, so let's not belittle the progress being made there.

Reply:
I see. so that is basically a problem if we are calling groupby getitem with size, groupby(['A', 'B']).['A'].size(), however, for regular groupby('A').size() or series.groupby().size() should still working fine. Could you please leave a comment here in the code with the ticket number about why are we doing the correction here?

This could potentially impact other APIs also, we probably should mark it as a bug. Furthermore, this may not be something we want to follow modin's implementation, there seems a reason for us to choose iloc in particular.

Reply:
Sounds great, will add a TODO there.